compiler/lib/code.mli

```diff
@@ -185,6 +185,9 @@ type constant =
   | Tuple of int * constant array * array_or_not
   | Int of int32
 
+(** Guaranteed equality in terms of both OCaml [(=)] and Javascript [(==)]: if
+    [constant_equal a b = Some v], then [Poly.(=) a b = v] and
+    [Js_of_ocaml.Js.equals a b = v]. This is used for optimization purposes. *)
```

Comment:
I don't think the second part is correct. two strings, int64, tuple, or floatArray might be equal under constant_equal and different under Js_of_ocaml.Js.equals.
`caml_js_equals` in `eval_instr` looks incorrect.

Reply:
Can you give an example of two strings that would be equal under constant_equal and not under `Js.equals`?

Reply:
For strings, you need to set the option `use-js-string` to false.
You can try tuples: `Js.equals (0,0) (0, 0)` should be false.

Reply:
OK, it can be the case if the two values are structurally equal but are different objects.

Reply:
> You can try tuples: Js.equals (0,0) (0, 0) should be false.

Mm, `(0,0) == (0,0)` is true in NodeJS.

Reply:
`(0,0)` is not a syntax for tuples in JavaScript...
```
> [0,0,0]==[0,0,0]
false
```